ext/EnzymeStaticArraysExt.jl

```diff
@@ -32,11 +32,37 @@ end
     end
 end
 
-@inline function Enzyme.EnzymeCore.make_zero(x::FT)::FT where {FT<:SArray}
```

Comment:
Looks like the Sarray one arg special case was removed, can that be restored?

Reply:
Also marray (non in place)

Reply:
Yeah, had noticed the same but didn't want to interrupt CI again. Will push a fix. (Though these methods aren't strictly needed, as evidenced by the tests, but I suppose `zero` and `fill!` might be a bit more efficient than recursing into the structs.)

Reply:
They’re necessary for performance since otherwise Enzyme.gradient and the likes with an sarray recurses and generates a lot of unnecessary allocations (slowing things down)

Reply:
Ah, now I see why I removed them in the first place, the previous methods didn't constraint the eltype to floats/complex floats, which led to a lot of test failures. Anyway, proper methods added now.